cli/flox/src/commands/activate.rs

```diff
@@ -671,6 +691,46 @@ impl Activate {
     }
 }
 
+/// Notify the user of available upgrades
+///
+/// Upon activation flox will start a detached process to check for upgrades.
+/// Future activations will be able to read the upgrade information from a file
+/// and notify the user if there are any upgrades available using this function.
+fn notify_upgrade_if_available(flox: &Flox, environment: &mut ConcreteEnvironment) -> Result<()> {
+    let upgrade_guard = UpgradeInformationGuard::read_in(environment.cache_path()?)?;
```

Comment:
nonblocking: I think we should use flox.cache_dir for consistency with the Flox upgrade notification, and also because of the `flox activate -d something-non-writable` case. The downside is cleanup, so if that's too painful maybe we shouldn't worry about that for now (since we write logs to .flox anyways)

Reply:
Counterpoint: maybe we shouldn't print the notification if the user running the command can't run the upgrade